src/main/java/org/cascadebot/cascadebot/events/CommandListener.java

```diff
@@ -81,10 +82,17 @@ public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
         String trigger;
         String[] args;
 
+        boolean messagesBot = false;
+        for (Member member : event.getMessage().getMentionedMembers()) {
+            if (member.getIdLong() == event.getGuild().getSelfMember().getIdLong()) {
+                messagesBot = true;
+            }
+        }
+
         if (message.startsWith(prefix)) {
             commandWithArgs = message.substring(prefix.length()); // Remove prefix from command
-        } else if (guildData.getCoreSettings().isMentionPrefix() && message.startsWith(event.getJDA().getSelfUser().getAsMention())) {
-            commandWithArgs = message.substring(event.getJDA().getSelfUser().getAsMention().length()).trim();
+        } else if (guildData.getCoreSettings().isMentionPrefix() && messagesBot) {
+            commandWithArgs = message.substring(event.getJDA().getSelfUser().getAsMention().length() + 1).trim();
```

Comment:
I'm *pretty* sure that if we use context.getMember().getAsMention() this will work as before.